netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -472,7 +473,11 @@ private void sendPingFrame(ChannelHandlerContext ctx, SendPingCommand msg,
       @Override
       public void operationComplete(ChannelFuture future) throws Exception {
         if (!future.isSuccess()) {
-          finalPing.failed(future.cause());
+          Throwable cause = future.cause();
+          if (cause instanceof ClosedChannelException) {
+            cause = lifecycleManager.getShutdownThrowable();
```

Comment:
Then null is propagated to the `PingCallback`. The API spec seems to allow it. If our code works correctly, we should never get a `ClosedChannelException` without the `ClientTransportLifeCycleManager` knowing about it, no?